src/elements/Label/Label.js

```diff
@@ -0,0 +1,168 @@
+/* eslint-disable valid-jsdoc, complexity */
+import cx from 'classnames'
+import React, { createElement, PropTypes } from 'react'
+import createFragment from 'react-addons-create-fragment'
+
+import META from '../../utils/Meta'
+import * as sui from '../../utils/semanticUtils'
+import { someChildType } from '../../utils/childrenUtils'
+import {
+  iconPropRenderer,
+  imagePropRenderer,
+  useKeyOnly,
+  useValueAndKey,
+  useKeyOrValueAndKey,
+} from '../../utils/propUtils'
+import Icon from '../Icon/Icon'
+import Image from '../Image/Image'
+
+/**
+ * A label displays content classification
+ */
+const Label = (props) => {
+  const {
+    attached, children, color, corner, className, circular, detail, detailLink, floating, horizontal, icon,
+    image, link, onClick, onClickDetail, onClickRemove, pointing, removable, ribbon, size, tag, text,
+    ...rest,
+  } = props
+
+  const handleClick = e => onClick && onClick(e, props)
+  const handleClickRemove = e => onClickRemove && onClickRemove(e, props)
+  const handleClickDetail = e => onClickDetail && onClickDetail(e, props)
+
+  const _component = image || link || onClick ? 'a' : 'div'
+
+  const classes = cx('sd-label ui',
+    size,
+    color,
+    useKeyOnly(floating, 'floating'),
+    useKeyOnly(horizontal, 'horizontal'),
+    useKeyOnly(pointing, 'pointing'),
+    useKeyOnly(tag, 'tag'),
+    useValueAndKey(attached, 'attached'),
+    useValueAndKey(corner, 'corner'),
+    useKeyOrValueAndKey(pointing, 'pointing'),
+    useKeyOrValueAndKey(ribbon, 'ribbon'),
```

Comment:
I really struggled with these, really.  Would love any input, I want it to be intuitive and clear.  The backstory is spelled out in [this comment](https://github.com/TechnologyAdvice/stardust/pull/215#discussion_r58302514) above.

The alternative is to write these out with no function helpers.  Slightly more performant, but more error prone and duplicates a lot of logic.  Every component will have a stack of these "prop to className" transformations.

Here is what it looks like without the helper functions:

```js
const classes = cx('sd-label ui',
  size,
  color,
  // key only
  floating && 'floating',
  horizontal && 'horizontal',
  pointing && 'pointing',
  tag && 'tag',
  // value and key
  attached && attached !== true && `${attached} ${'attached'}`,
  corner && corner !== true && `${corner} ${'corner'}`,
  // key only || value and key
  pointing && (pointing === true ? 'pointing' : `${pointing} ${'pointing'}`),
  ribbon && (ribbon === true ? 'ribbon' : `${ribbon} ${'ribbon'}`),
  // ...others
  circular && (children && 'circular' || 'empty circular'),
  // TODO how to handle image child with no image class? there are two image style labels.
  (image || someChildType(children, Image) || someChildType(children, 'img')) && 'image',
  'label',
  className
)
```

Thoughts or ideas welcome.